### PR TITLE
Auth Documentation

### PR DESCRIPTION
**Pull request by Zatine**
# Hey, I just made a Pull Request!

## Description

Added documentation in the API and readme about the newly introduced Authorization functionality.

## Checklist for PR author(s)

- [ ] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [x] Relevant documentation updated
- [x] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -197,8 +197,11 @@ Styx will refuse to trigger the workflow.
 
 In order for Styx to be able to create/delete keys for the `service_account` of a workflow,
 the [Service Account] that Styx itself runs as should be granted `Service Account Key Admin`
-role for the `service_account` of the workflow. This can be done by following
-[Granting Roles to Service Accounts].
+role for the `service_account` of the workflow.
+
+If authorization is enabled for the service, the `service_account` will be used to authorize deployments and actions on the workflow. To authorize an account, grant it the [configured role](./styx-standalone-service/src/main/resources/styx-standalone.conf#L66) for the [Service Account] of the workflow.
```
Comment (honnix) [inline README.md]: We can elaborate a bit more about `actions`.

> create/modify/delete a workflow, trigger a new instance, retry/halt an existing instance and create backfill

**Comment by honnix** [comment]
minor comment. please break the long lines.

**Comment by Zatine** [comment]
@honnix please take another look

File: README.md [hunk 2]
```diff
@@ -300,3 +307,6 @@ expected to honor this code.
 [codecov.io]: https://codecov.io/gh/spotify/styx
 [styx-api-service]: https://github.com/spotify/styx/tree/master/styx-api-service
 [styx-scheduler-service]: https://github.com/spotify/styx/tree/master/styx-scheduler-service
+[configured role]: ./styx-standalone-service/src/main/resources/styx-standalone.conf#L66
```
Comment (honnix) [inline README.md]: ⭐️

**Pull request merged**